File: android/app/src/main/java/my/ssdid/wallet/domain/oid4vp/PresentationSubmission.kt
package my.ssdid.wallet.domain.oid4vp

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

/**
 * Presentation Exchange 2.0 response metadata.
 *
 * Describes how the VP Token maps to the requested Presentation Definition.
 */
@Serializable
data class PresentationSubmission(
    val id: String,
    @SerialName("definition_id") val definitionId: String,
    @SerialName("descriptor_map") val descriptorMap: List<DescriptorMapEntry>
) {
    fun toJson(): String = Json.encodeToString(serializer(), this)
}

@Serializable
data class DescriptorMapEntry(
    val id: String,
    val format: String,
    val path: String
)

File: ios/SsdidWallet/Domain/Oid4vp/PresentationSubmission.swift
import Foundation

struct PresentationSubmission: Codable {
    let id: String
    let definitionId: String
    let descriptorMap: [DescriptorMapEntry]

    enum CodingKeys: String, CodingKey {
        case id
        case definitionId = "definition_id"
        case descriptorMap = "descriptor_map"
    }

    func toJson() throws -> String {
        let data = try JSONEncoder().encode(self)
        return String(data: data, encoding: .utf8) ?? "{}"
    }
}

struct DescriptorMapEntry: Codable {
    let id: String
    let format: String
    let path: String
}
